plugins/statistics.koplugin/main.lua

```diff
@@ -753,12 +754,28 @@ function ReaderStatistics:getIdBookDB()
         SELECT count(id)
         FROM   book
         WHERE  title = ?
-            AND authors = ?
-            AND md5 = ?;
+          AND  authors = ?
+          AND  md5 = ?;
     ]]
     local stmt = conn:prepare(sql_stmt)
-    local result = stmt:reset():bind(self.data.title, self.data.authors, self.doc_md5):step()
+    local title, authors = self.data.title, self.data.authors
+    local result = stmt:reset():bind(title, authors, self.doc_md5):step()
     local nr_id = tonumber(result[1])
+    if nr_id == 0 and self.ui.paging then -- old strings are null-terminated
+        title = title .. "\0"
+        result = stmt:reset():bind(title, authors, self.doc_md5):step()
+        nr_id = tonumber(result[1])
+        if nr_id == 0 then
+            authors = authors .. "\0"
+            result = stmt:reset():bind(title, authors, self.doc_md5):step()
+            nr_id = tonumber(result[1])
+            if nr_id == 0 then
+                title = self.data.title
+                result = stmt:reset():bind(title, authors, self.doc_md5):step()
+                nr_id = tonumber(result[1])
+            end
+        end
+    end
```

Comment:
Can we *really* get a mismatch between authors & titles in terms of NUL termination (e.g., only *one* of them being "sane")?

Not familiar with the whole thing, but I would just have basically expected  a single fallback to `result = stmt:reset():bind(title .. '\0', authors .. '\0', self.doc_md5):step()`

Reply:
I think yes. If document title is absent, we use display_title (filename without extension), if document authors is absent, we use "N/A".

Reply:
I've checked once more.
If a property is absent, muPDF returns "" (empty string without null-terminator).

Then we do
https://github.com/koreader/koreader/blob/ca569c1355d14ad023eb1bcad7b995ae027e423d/frontend/document/document.lua#L175-L177
So, in `doc_props` we have `nil` for empty properties.
Then we build `display_title` (that doesn't have null-terminator if `title` is empty).
https://github.com/koreader/koreader/blob/ca569c1355d14ad023eb1bcad7b995ae027e423d/frontend/apps/filemanager/filemanagerbookinfo.lua#L218
Finally the statistics plugin uses:
https://github.com/koreader/koreader/blob/ca569c1355d14ad023eb1bcad7b995ae027e423d/plugins/statistics.koplugin/main.lua#L176-L177
Summarizing: old `authors` and `title` in stats had null-terminator if they existed, and didn't have if any of them was empty in the document. We need to check 4 combinations.